### PR TITLE
feat(toaster): improve toast components

### PR DESCRIPTION
**Pull request by larsdouweschuitema**
- make animationDuration a dynamic property that can be set
- change default duration to 5 seconds instead of endless 15 seconds
- change close button to always show rather than on hover which causes issues on mobile and desktop
- write unit tests for toaster and toast components

**Dynamic property**
We noticed on the new search that the duration is too long and continious actions then end up stacking toasts. We believe 5 seconds is a better default, but we still want to give you the option to change it, because some messages might be longer and require for longer durations.

**Close button**
The close button only shows on hover on the toast which gives the user the impression there is no close button. On mobile you can't hover and you would need to press on the component to see any close button. We decided showing the close button by default is the best user experience.

**Missing unit tests**
These components had no unit tests so I added some tests which are the most important for the library to test.

### CODE REVIEW

**Comment by pijnenburger** [comment]
Agree that the duration is too long & that it should be a dynamic property with a good default. Also think always showing a close button or making the whole container be the close button is a good idea.

But toasts are made to be stackable btw, just our version of the toast is quite massive, not optimized for stacking and is being used for error messages now (which is a bad practice).

For now I think the changes are at least an improvement but good for the design team at least to look at again and see if it's used properly, with the right goals in mind.

**Comment by larsdouweschuitema** [comment]
> Agree that the duration is too long & that it should be a dynamic property with a good default. Also think always showing a close button or making the whole container be the close button is a good idea.
> 
> But toasts are made to be stackable btw, just our version of the toast is quite massive, not optimized for stacking and is being used for error messages now (which is a bad practice).
> 
> For now I think the changes are at least an improvement but good for the design team at least to look at again and see if it's used properly, with the right goals in mind.

@ConnorChristensen I think you misunderstood. The toaster is still stackable, but my point was: the toasts don't dissapear with a duration of 15 seconds and I only reduced the duration, to better fit the user.

File: src/components/ui-toaster.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
             enter-active-class="animate-[ui-fade-right_0.6s_1_ease-out]"
             leave-active-class="animate-[ui-fade-right_0.6s_1_ease-out_reverse]"
         >
-            <ui-toast
+            <UiToast
```
[the remark this answers is not in the document]
Comment (larsdouweschuitema) [reply src/components/ui-toaster.vue:8]: No

**Review by anilkusaksiz — APPROVED** [review]

**Comment by larsdouweschuitema** [comment]
- [ ] Update story with missing message

**Review by anilkusaksiz — APPROVED** [review]

**Review by pijnenburger — APPROVED** [review]

**Comment by ConnorChristensen** [comment]
@larsdouweschuitema These changes are good, but I think you should split this into different commits. If you merge this in as is, the changelog will have a single feature entry for "improve toast components", which is not descriptive enough to know what changed without checking the commit message.

**Comment by larsdouweschuitema** [comment]
> @larsdouweschuitema These changes are good, but I think you should split this into different commits. If you merge this in as is, the changelog will have a single feature entry for "improve toast components", which is not descriptive enough to know what changed without checking the commit message.

- [ ]  improve commit messages

**Review by ConnorChristensen — APPROVED** [review]
Nice work!

**Comment by larsdouweschuitema** [comment]
> Nice work!

Thank you Sir!!!

**Pull request merged**